pkg/cmd/openshift-service-serving-cert-signer/config.go

```diff
@@ -0,0 +1,214 @@
+
+func readYAMLFileInto(filename string, obj runtime.Object) error {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	err = readYAMLInto(data, obj)
+	if err != nil {
+		return fmt.Errorf("could not load config file %q due to an error: %v", filename, err)
+	}
+	return nil
+}
+
+// TODO: we ultimately want a better decoder for JSON that allows us exact line numbers and better
+// surrounding text description. This should be removed / replaced when that happens.
+func captureSurroundingJSONForError(prefix string, data []byte, err error) error {
+	if syntaxErr, ok := err.(*json.SyntaxError); err != nil && ok {
+		offset := syntaxErr.Offset
+		begin := offset - 20
+		if begin < 0 {
+			begin = 0
+		}
+		end := offset + 20
+		if end > int64(len(data)) {
+			end = int64(len(data))
+		}
+		return fmt.Errorf("%s%v (found near '%s')", prefix, err, string(data[begin:end]))
+	}
+	if err != nil {
+		return fmt.Errorf("%s%v", prefix, err)
+	}
+	return err
+}
+
+func resolveConfigPaths(config *v1alpha1.ServiceServingCertSignerConfig, base string) error {
+	return resolvePaths(getFileReferences(config), base)
+}
+
+func getFileReferences(config *v1alpha1.ServiceServingCertSignerConfig) []*string {
+	refs := []*string{}
+
+	refs = append(refs, &config.ServingInfo.CertInfo.CertFile)
+	refs = append(refs, &config.ServingInfo.CertInfo.KeyFile)
+	refs = append(refs, &config.ServingInfo.ClientCA)
+	for i := range config.ServingInfo.NamedCertificates {
+		refs = append(refs, &config.ServingInfo.NamedCertificates[i].CertFile)
+		refs = append(refs, &config.ServingInfo.NamedCertificates[i].KeyFile)
+	}
+	refs = append(refs, &config.Signer.CertFile)
+	refs = append(refs, &config.Signer.KeyFile)
+
+	return refs
+}
+
+// resolvePaths updates the given refs to be absolute paths, relative to the given base directory
+func resolvePaths(refs []*string, base string) error {
+	for _, ref := range refs {
+		// Don't resolve empty paths
+		if len(*ref) > 0 {
+			// Don't resolve absolute paths
+			if !filepath.IsAbs(*ref) {
+				*ref = filepath.Join(base, *ref)
+			}
+		}
+	}
+	return nil
+}
+
+type ClientConnectionOverrides struct {
+	// AcceptContentTypes defines the Accept header sent by clients when connecting to a server, overriding the
+	// default value of 'application/json'. This field will control all connections to the server used by a particular
+	// client.
+	AcceptContentTypes string
+	// ContentType is the content type used when sending data to the server from this client.
+	ContentType string
+
+	// QPS controls the number of queries per second allowed for this connection.
+	QPS float32
+	// Burst allows extra queries to accumulate when a client is exceeding its rate.
+	Burst int32
+}
+
+var defaultConnectionOverrides = &ClientConnectionOverrides{
+	AcceptContentTypes: "application/vnd.kubernetes.protobuf,application/json",
+	ContentType:        "application/vnd.kubernetes.protobuf",
+	QPS:                50,
+	Burst:              100,
```

Comment:
It's much clearer to define default values as consts at the top of file, rather than buried farther down. Also, please add a comment about why these are reasonable defaults.